src/shared/dropdown-list-v2/dropdown-list.styles.tsx

```diff
@@ -144,6 +145,13 @@ export const CheckboxUnselectedIndicator = styled(SquareIcon)`
     color: ${Colour["icon-primary-subtlest"]};
 `;
 
+export const CheckboxDisabledIndicator = styled(SquareFillIcon)`
+    flex-shrink: 0;
+    height: 1lh;
+    width: 1lh;
+    color: ${Colour["icon-disabled"]};
```

Comment:
we can go with `icon-disabled-subtle` to match the Checkbox component - it's a lighter shade of gray